gamemodes/terrortown/gamemode/server/sv_shop.lua

```diff
@@ -16,138 +16,147 @@ end
 
-		credits = equip_table.credits or 1
+	local credits = equip_table.credits or 1
 
-		if credits > ply:GetCredits() then
-			print(ply, "tried to buy item/weapon, but didn't had enough credits")
+	if credits > ply:GetCredits() then
+		print(ply, "tried to buy item/weapon, but didn't have enough credits")
 
-			return
-		end
+		return
+	end
 
-		if GetGlobalInt("ttt2_random_shops") > 0 and RANDOMSHOP[ply] and #RANDOMSHOP[ply] > 0 then
-			local key = false
+	-- keep compatibility with old addons
+	if not hook.Run("TTTCanOrderEquipment", ply, idOrCls, is_item) then return end
 
-			for i = 1, #RANDOMSHOP[ply] do
-				if RANDOMSHOP[ply][i].id ~= id then continue end
+	-- add our own hook with more consistent class parameter and some more information
+	local allow, ignoreCost, message = hook.Run("TTT2CanOrderEquipment", ply, cls, is_item, credits)
 
-				key = true
+	if message then
+		LANG.Msg(ply, message, nil, MSG_MSTACK_ROLE)
+	end
 
-				break
-			end
+	if allow == false then
+		return
+	end
 
-			if not key then
-				print(ply, "tried to buy item/weapon, but didn't allowed to do so!")
+	if not ignoreCost then
+		ply:SubtractCredits(credits)
+	end
 
-				return
-			end
-		end
+	ply:AddBought(cls)
 
-		-- no longer restricted to only WEAPON_EQUIP weapons, just anything that
-		-- is whitelisted and carryable
-		if not is_item then
-			if ply:CanCarryWeapon(equip_table) then
-				ply:GiveEquipmentWeapon(id)
-			else
-				return
-			end
-		else
-			ply:GiveEquipmentItem(id)
-		end
+	-- no longer restricted to only WEAPON_EQUIP weapons, just anything that
+	-- is whitelisted and carryable
+	if is_item then
+		ply:GiveEquipmentItem(cls)
 	else
-		print(ply, "tried to buy equip that doesn't exists", id)
-
-		return
+		ply:GiveEquipmentWeapon(cls)
 	end
 
-	ply:SubtractCredits(credits)
-
 	LANG.Msg(ply, "buy_received", nil, MSG_MSTACK_ROLE)
 
-	ply:AddBought(id)
-
 	timer.Simple(0.5, function()
 		if not IsValid(ply) then return end
 
 		net.Start("TTT_BoughtItem")
-		net.WriteString(id)
+		net.WriteString(cls)
 		net.Send(ply)
 	end)
 
-	if is_item and equip_table and isfunction(equip_table.Bought) then
+	if isfunction(equip_table.Bought) then
 		equip_table:Bought(ply)
 	end
 
-	-- still support old items
-	local id2 = (is_item and equip_table.oldId or nil) or id
-
-	hook.Call("TTTOrderedEquipment", GAMEMODE, ply, id2, is_item and id2 or false)
-
 	if GetGlobalBool("ttt2_random_shop_reroll_per_buy") then
 		RerollShop(ply)
 	end
+
+	-- keep compatibility with old addons
+	hook.Run("TTTOrderedEquipment", ply, idOrCls, is_item)
+
+	-- add our own hook with more consistent class parameter
+	hook.Run("TTT2OrderedEquipment", ply, cls, is_item, credits, ignoreCost or false)
+end
+
+local function NetOrderEquipment(len, ply)
+	local cls = net.ReadString()
+
+	if #cls == 0 then return end
+
+	OrderEquipment(ply, cls)
 end
-concommand.Add("ttt_order_equipment", OrderEquipment)
+
+local function ConOrderEquipment(ply, cmd, args)
```

Comment:
The function name could be read misleadingly as "CanOrderEquipment" (at least I stuck a moment here). Maybe change it to ConcommandOrderEquipment or something similar.

Reply:
I agree. Ill change it to ConCommandOrderEquipment

Reply:
Done 06aa3db